common/matrix/ell_kernels.hpp.inc

```diff
@@ -46,19 +46,22 @@ __device__ void spmv_kernel(const size_type num_rows, const int nwarps_per_row,
                             const size_type c_stride, Closure op)
 {
     const auto tidx =
-        static_cast<IndexType>(blockDim.x) * blockIdx.x + threadIdx.x;
-    const IndexType x = tidx / subwarp_size / nwarps_per_row;
-    const auto warp_id = tidx / subwarp_size % nwarps_per_row;
-    const auto y_start = tidx % subwarp_size +
-                         num_stored_elements_per_row * warp_id / nwarps_per_row;
-    const auto y_end =
-        num_stored_elements_per_row * (warp_id + 1) / nwarps_per_row;
-    if (x < num_rows) {
-        const auto tile_block =
-            group::tiled_partition<subwarp_size>(group::this_thread_block());
+        static_cast<size_type>(blockDim.x) * blockIdx.x + threadIdx.x;
```

Comment:
nit: 
```suggestion
        size_type(blockDim.x) * blockIdx.x + threadIdx.x;
```

Reply:
I would actually prefer `static_cast` here since it is in fact a cast.
Usually, I would use `size_type(12)` with actual numbers (for me, this is using a constructor vs. casting).

Since I don't think we have any guidelines for it, feel free to use what you want @yhmtsai, since both options should result in the exact same behavior in this case.

Reply:
I would agree that this is a matter of taste. My reasons for proposing all these changes are
1. they are semantically equivalent (except for composite type names like `unsigned int`, they can be used interchangably) and
2. the "constructor" solution is shorter.

I would however not agree that there is a difference between construction and conversion (i.e., casting) for primitive types. Maybe we can discuss this question in general and possibly find a common guideline before I start spamming all pull requests with these change suggestions ;)

Reply:
For primitive types, you are correct, there is no difference. However, for more complex types, it can make a difference.
I just like to have an explicit `cast` word, so everyone knows you are casting here.

I think adding either solution to the guidelines might be a bit too detailed since the guideline page is currently quite long. But we can discuss this tomorrow.

Reply:
For more complex types, I'm not so sure there is even a difference. See Section 5.2.3:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2011/n3242.pdf

If I understand all of this properly, this notation would even dynamic cast/reinterpret cast transparently? I never tested this but they link to section 5.4 which is for all casting operations.
We are going into the details of C++ here though.

Reply:
Also, a minor detail: in C++ standard jargon, they rather call this type of cast "functional notation" rather than "constructor notation".